### PR TITLE
[Enhancement](log) Improve Safety and Robustness of Log4j Configuration

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by liugddx — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/common/Log4jConfig.java [hunk 1]
```diff
@@ -228,11 +229,22 @@ private static void reconfig() throws IOException {
...
+            throw new IOException("The configuration template is empty!");
+        }
+
+        Log4jConfig config;
+        try (ByteArrayInputStream bis = new ByteArrayInputStream(newXmlConfTemplate.getBytes(StandardCharsets.UTF_8))) {
+            ConfigurationSource source = new ConfigurationSource(bis);
+            config = new Log4jConfig(source);
+
+            LoggerContext context = (LoggerContext) LogManager.getContext(false);
+            context.start(config);
```
Comment (DongLiang-0) [inline fe/fe-core/src/main/java/org/apache/doris/common/Log4jConfig.java:242]: I see that `context.start(config)` is called on both lines 248 and 242. Do we need to call start twice here?

**Comment by zy-kkk** [comment]
run buildall

**Review by CalvinKirs — APPROVED** [review]

**Review by DongLiang-0 — APPROVED** [review]

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**